File: app/src/main/java/com/ozkanmut/ilactakip/NtfyEndpoint.kt
package com.ozkanmut.ilactakip

import java.net.URLEncoder

object NtfyEndpoint {
    const val BASE_URL = "https://ntfy.field-maintenance-prod.com"

    fun topicUrl(topic: String): String = "$BASE_URL/${encode(topic)}"

    fun pollUrl(topics: List<String>, since: String): String {
        val topicPath = topics.joinToString(",") { encode(it) }
        return "$BASE_URL/$topicPath/json?poll=1&since=${encode(since)}"
    }

    fun streamUrl(topics: List<String>, since: String): String {
        val topicPath = topics.joinToString(",") { encode(it) }
        return "$BASE_URL/$topicPath/json?since=${encode(since)}"
    }

    private fun encode(value: String): String = URLEncoder.encode(value, "UTF-8")
}

File: ios/Dosefolk/Dosefolk/NtfyEndpoint.swift
import Foundation

enum NtfyEndpoint {
    static let baseURL = "https://ntfy.field-maintenance-prod.com"

    static func topicURL(_ topic: String) -> URL? {
        URL(string: "\(baseURL)/\(encode(topic))")
    }

    static func pollURL(topics: [String], since: String) -> URL? {
        let path = topics.map(encode).joined(separator: ",")
        return URL(string: "\(baseURL)/\(path)/json?poll=1&since=\(encode(since))")
    }

    static func streamURL(topics: [String], since: String) -> URL? {
        let path = topics.map(encode).joined(separator: ",")
        return URL(string: "\(baseURL)/\(path)/json?since=\(encode(since))")
    }

    private static func encode(_ value: String) -> String {
        var allowed = CharacterSet.alphanumerics
        allowed.insert(charactersIn: "-._~")
        return value.addingPercentEncoding(withAllowedCharacters: allowed) ?? ""
    }
}
